src/ast/passes/codegen_llvm.h

```diff
@@ -280,19 +274,43 @@ class CodegenLLVM : public Visitor {
 
   std::map<std::string, AllocaInst *> variables_;
 
-  // NB: ensure all IDs are saved/restored in create_reset_ids()
-  int printf_id_ = 0;
-  int mapped_printf_id_ = 0;
-  int time_id_ = 0;
-  int cat_id_ = 0;
-  int strftime_id_ = 0;
-  uint64_t join_id_ = 0;
-  int system_id_ = 0;
-  int non_map_print_id_ = 0;
-  uint64_t watchpoint_id_ = 0;
-  int cgroup_path_id_ = 0;
-  int skb_output_id_ = 0;
-  int str_id_ = 0;
+#define FOR_LIST_OF_ASYNC_IDS(DO)                                              \
```

Comment:
IMHO kinda ugly to have all these macros defined inline. Is there a way to move the macro magic out into a separate file / class? Then in here would could store just a `AsyncIds async_ids_` member. Callers could do `async_ids_.printf()` or something like that.

Reply:
Yeah I thought about a separate class too. I think I just got lazy 😄  - I'll update the PR